src/app/search/loadouts/search-filters/freeform.ts

```diff
@@ -22,6 +44,49 @@ const freeformFilters: FilterDefinition<
       return (loadout) => test(loadout.name);
     },
   },
+  {
+    keywords: ['subclass'],
+    description: tl('LoadoutFilter.Subclass'),
+    format: 'freeform',
+    suggestionsGenerator: ({ loadouts, d2Definitions }) => {
+      if (!loadouts || !d2Definitions) {
+        return [];
+      }
+      const damageDefs = damageDefsByDamageType(d2Definitions);
+      // TODO (ryan) filter on currently selected character. This info is currently localized
+      // to the page, so we need to lift that up before it can be done.
+      return deduplicate(
+        loadouts.flatMap((loadout) => {
+          const subclass = subclassDefFromLoadout(loadout, d2Definitions);
+          const damageType = subclass ? getDamageTypeForSubclassDef(subclass) : undefined;
+          // DamageType.None is 0
+          const damageName =
+            damageType !== undefined ? damageDefs[damageType]?.displayProperties.name : undefined;
+          return [
+            subclass
+              ? `subclass:${quoteFilterString(subclass.displayProperties.name.toLowerCase())}`
+              : undefined,
+            damageName ? `subclass:${quoteFilterString(damageName.toLowerCase())}` : undefined,
+          ];
+        }),
+      );
+    },
+    filter: ({ filterValue, language, d2Definitions }) => {
+      const test = matchText(filterValue, language, false);
+      const damageDefs = d2Definitions && damageDefsByDamageType(d2Definitions);
+      return (loadout: Loadout) => {
+        const subclass = d2Definitions ? subclassDefFromLoadout(loadout, d2Definitions) : undefined;
+        // DamageType.None is 0
+        const damageType = subclass ? getDamageTypeForSubclassDef(subclass) : undefined;
```

Comment:
I feel like a strategic early return could save you some ternaries here. You can also assume that if you have a subclass it has a damage type.

Reply:
Yeah I considered it at the time so happy to